ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,435 @@
+    pub slot_index: u8,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum OriginConsole {
+    ConsoleWii,
+    ConsoleDSi,
+    Console3DS,
+    ConsoleWiiUSwitch
+}
+
+#[derive(Clone, Debug)]
+pub struct ConsoleIdentity {
+    pub origin_console: OriginConsole,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum MiiSex {
+    Male,
+    Female,
+}
+
+#[derive(Clone, Debug)]
+pub struct Details {
+    pub sex: MiiSex,
+    pub birthday_month: u8,
+    pub birthday_day: u8,
+    pub shirt_color: u8,
+    pub is_favorite: bool,
+}
+
+#[derive(Clone, Debug)]
+pub struct FaceStyle {
+    pub is_sharing_enabled: bool,
+    pub shape: u8,
+    pub skin_color: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct FaceDetails {
+    pub wrinkles: u8,
+    pub makeup: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct HairDetails {
+    pub color: u8,
+    pub is_flipped: bool,
+}
+
+#[derive(Clone, Debug)]
+pub struct EyeDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_scale: u8,
+    pub rotation: u8,
+    pub x_spacing: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct EyebrowDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_scale: u8,
+    pub rotation: u8,
+    pub x_spacing: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct NoseDetails {
+    pub style: u8,
+    pub scale: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MouthDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_scale: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MustacheDetails {
+    pub mouth_y_position: u8,
+    pub mustache_style: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct BeardDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct GlassDetails {
+    pub style: u8,
+    pub color: u8,
+    pub scale: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MoleDetails {
+    pub is_enabled: bool,
+    pub scale: u8,
+    pub x_position: u8,
+    pub y_position: u8,
+}
+
+#[derive(Clone, Debug)]
+pub struct MiiData {
+    pub mii_options: MiiDataOptions,
+    pub mii_selector_position: SelectorPosition,
+    pub mii_console_identity: ConsoleIdentity,
+
+    pub system_id: [u8; 8],
+    pub mac_address: [u8; 6],
+
+    pub mii_details: Details,
+    pub mii_name: String,
+
+    pub height: u8,
+    pub width: u8,
+
+    pub face_style: FaceStyle,
+    pub face_details: FaceDetails,
+
+    pub hair_style: u8,
+
+    pub hair_details: HairDetails,
+    pub eye_details: EyeDetails,
+    pub eyebrow_details: EyebrowDetails,
+    pub nose_details: NoseDetails,
+    pub mouth_details: MouthDetails,
+    pub mustache_details: MustacheDetails,
+    pub beard_details: BeardDetails,
+    pub glass_details: GlassDetails,
+    pub mole_details: MoleDetails,
+
+    pub author_name: String,
+}
+
+impl From<[u8; 92]> for MiiData {
```

Comment:
Use the `ctru-sys` type rather than `[u8;92]`. Do this for all the cases in which it appears.

Reply:
The problem is that [there is no ``ctru-sys`` type for the ``[u8; 92]``](https://github.com/rust3ds/ctru-rs/blob/9cf5e4de94e8d3711fd757e3464dd1880d91b92f/ctru-sys/src/bindings.rs#L17597), but I could make a ype alias